2_dkr.py

```diff
@@ -0,0 +1,348 @@
+    """
+    Allow the user to replace all copies of one letter in the hand (chosen by user)
+    with a new letter chosen from the VOWELS and CONSONANTS at random. The new letter
+    should be different from user's choice, and should not be any of the letters
+    already in the hand.
+
+    If user provide a letter not in the hand, the hand should be the same.
+
+    Has no side effects: does not mutate hand.
+
+    For example:
+        substitute_hand({'h':1, 'e':1, 'l':2, 'o':1}, 'l')
+    might return:
+        {'h':1, 'e':1, 'o':1, 'x':2} -> if the new letter is 'x'
+    The new letter should not be 'h', 'e', 'l', or 'o' since those letters were
+    already in the hand.
+
+    hand: dictionary (string -> int)
+    letter: string
+    returns: dictionary (string -> int)
+    """
+    try:
+        letter_to_be_changed = hand[letter]
+    except:
+        return hand
+
+    while True:
+        new = random.choice(string.ascii_letters.lower())
+        if new not in hand:
+            break
+        continue
+
+    del hand[letter]
+    hand.update({new: letter_to_be_changed})
+    return hand
+
+
+def play_game(word_list):
+    """
+    Allow the user to play a series of hands
+
+    * Asks the user to input a total number of hands
+
+    * Accumulates the score for each hand into a total score for the
+      entire series
+
+    * For each hand, before playing, ask the user if they want to substitute
+      one letter for another. If the user inputs 'yes', prompt them for their
+      desired letter. This can only be done once during the game. Once the
+      substitue option is used, the user should not be asked if they want to
+      substitute letters in the future.
+
+    * For each hand, ask the user if they would like to replay the hand.
+      If the user inputs 'yes', they will replay the hand and keep
+      the better of the two scores for that hand.  This can only be done once
+      during the game. Once the replay option is used, the user should not
+      be asked if they want to replay future hands. Replaying the hand does
+      not count as one of the total number of hands the user initially
+      wanted to play.
+
+            * Note: if you replay a hand, you do not get the option to substitute
+                    a letter - you must play whatever hand you just had.
+
+    * Returns the total score for the series of hands
+
+    word_list: list of lowercase strings
+    """
+
+    all_points = 0
+    print("Let's start the game!")
+    number_of_hands = int(input("Input the total number of hands: "))
+    while number_of_hands != 0:
+        hand = deal_hand(hand_size)
+        print("Current hand: ", end=" ")
+        display_hand(hand)
+        answer = str(input("Would you like to substitute a letter? "))
+        if answer == "yes":
+            r = str(input("Choose the letter to be changed, please: \n"))
+            hand = substitute_hand(hand, r)
+        else:
+            print("We continue!")

```

Comment:
According to script of homework there is no "We continue!" string. Did you discuss about this with Victor?